src/main/java/org/jabref/logic/importer/fileformat/BibtexParser.java

```diff
@@ -386,6 +390,20 @@ private void parseJabRefComment(Map<String, String> meta) {
             } catch (ParseException ex) {
                 parserResult.addException(ex);
             }
+        } else if (comment.substring(0, Math.min(comment.length(), MetaData.META_FLAG_010.length())).equals(MetaData.META_FLAG_010)) {
+            parseCommentToJson(comment, meta);
+        }
+    }
+
+    private void parseCommentToJson(String comment, Map<String, String> meta) {
```

Comment:
No - it should return the JSOn object.

The following call should be rewritten:

```java
            MetaData metaData = metaDataParser.parse(
                    meta,
                    importFormatPreferences.bibEntryPreferences().getKeywordSeparator());
```

`meta` should be a record of

- `Map<String, String> v5metaData`
- `JsonObject v6metaData`

And then be handled accordingliny in `org.jabref.logic.importer.fileformat.BibtexParser#metaDataParser`.